routes/recent.go

```diff
@@ -8,9 +8,45 @@ import (
 // Recent serves GET /recent
 func Recent(w http.ResponseWriter, r *http.Request) {
 	rows, err := db.Query(
-		` SELECT hole, lang, login, LENGTH(code), submitted
-		    FROM solutions JOIN users ON id = user_id
-		ORDER BY submitted DESC LIMIT 100`,
+		`WITH solution_lengths AS (
+	        SELECT hole,
+	               lang,
+	               login,
+	               LENGTH(code) strokes,
+	               submitted
+	          FROM solutions
+	          JOIN users on user_id = id
+	          WHERE NOT failing
+	    ) SELECT t1.hole,
+	             t1.lang,
+	             login,
+	             t1.strokes,
+	             min_strokes,
+	             COUNT(*) tie_count,
+	             t1.submitted
+	        FROM solution_lengths AS t1
+	  INNER JOIN (
+	          SELECT hole, lang, MIN(strokes) min_strokes
+	          FROM solution_lengths
+	          GROUP BY lang, hole
+	        ) AS t2
+	          ON t1.hole = t2.hole
+	         AND t1.lang = t2.lang
+	  INNER JOIN (
+	          SELECT hole, lang, strokes, submitted
+	          FROM solution_lengths
+	        ) AS t3
+	          ON t1.hole = t3.hole
+	         AND t1.lang = t3.lang
+	         AND t1.strokes = t3.strokes
+	         AND t1.submitted >= t3.submitted
+	    GROUP BY t1.hole,
+	             t1.lang,
+	             login,
+	             t1.strokes,
+	             min_strokes,
+	             t1.submitted
+	    ORDER BY t1.submitted DESC LIMIT 100`,
```

Comment:
This SQL is a bit complicated, mainly due to determining whether a solution set a new record. There may be simpler ways to do this.